install/kubernetes/cilium/templates/cilium-envoy/service.yaml

```diff
@@ -0,0 +1,27 @@
+{{- if and .Values.envoy.enabled (not .Values.preflight.enabled) .Values.envoy.prometheus.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: cilium-envoy
+  namespace: {{ .Release.Namespace }}
+  {{- if not .Values.envoy.prometheus.serviceMonitor.enabled }}
+  annotations:
+    prometheus.io/scrape: "true"
+    prometheus.io/port: {{ .Values.envoy.prometheus.port | quote }}
+  {{- end }}
+  labels:
+    k8s-app: cilium-envoy
+    app.kubernetes.io/name: cilium-envoy
+    app.kubernetes.io/part-of: cilium
+    io.cilium/app: proxy
+spec:
+  clusterIP: None
```

Comment:
any reason to use headless service here ?

Reply:
Indeed, this is more a "bag of pods" than a service.